### DISCUSSION TITLE
Firefox | Reading manifest: Error processing content_security_policy.extension_pages: ‘script-src’ directive contains a forbidden http: protocol source

### DISCUSSION THREAD

**Discussion by JVariance** [aklinker1/vite-plugin-web-extension#160] [Q&A] [answered]
Any idea how I can solve this issue in Firefox with Manifest Version 3?

```
Reading manifest: Error processing content_security_policy.extension_pages: ‘script-src’ directive contains a forbidden http: protocol source
```

I'm not allowed to use the inline script.
It's working in Chrome though.

```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta http-equiv="X-UA-Compatible" content="IE=edge" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		<link rel="stylesheet" href="./popup.css" />
		<title>Popup</title>
	</head>

	<body></body>

	<script type="module" src="./popup.ts"></script>
</html>

```

**Comment by aklinker1** [comment, marked as answer]
This is a bug in Firefox. I've reported it here already: https://bugzilla.mozilla.org/show_bug.cgi?id=1864284

For now, dev mode doesn't work in Firefox while using MV3 because it's trying to add `http://localhost` to the manifest's CSP, which is required for HMR.

To test against Firefox MV3, build for production and load the extension by hand.
